src/adaptation/stepsize.jl

```diff
@@ -0,0 +1,117 @@
+######################
+### Mutable states ###
+######################
+
+mutable struct DAState{T<:AbstractFloat}
+    m     :: Int
+    ϵ     :: T
+    μ     :: T
+    x_bar :: T
+    H_bar :: T
+end
+
+function DAState(ϵ::AbstractFloat)
+    μ = computeμ(ϵ)
+    return DAState(0, ϵ, μ, 0.0, 0.0)
+end
+
+function computeμ(ϵ::AbstractFloat)
+    return log(10 * ϵ) # see NUTS paper sec 3.2.1
+end
+
+function reset!(dastate::DAState{T}) where {T<:AbstractFloat}
+    dastate.μ = computeμ(dastate.ϵ)
+    dastate.m = 0
+    dastate.x_bar = zero(T)
+    dastate.H_bar = zero(T)
+end
+
+mutable struct MSSState{T<:AbstractFloat}
+    ϵ :: T
+end
+
+################
+### Adapters ###
+################
+
+abstract type StepSizeAdapter <: AbstractAdapter end
+
+struct FixedStepSize{T<:AbstractFloat} <: StepSizeAdapter
+    ϵ :: T
+end
+
+function getss(fss::FixedStepSize)
+    return fss.ϵ
+end
+
+struct DualAveraging{T<:AbstractFloat} <: StepSizeAdapter
+  γ     :: T
+  t_0   :: T
+  κ     :: T
+  δ     :: T
+  state :: DAState{T}
+end
+
+reset!(da::DualAveraging) = reset!(da.state)
+
+function DualAveraging(γ::AbstractFloat, t_0::AbstractFloat, κ::AbstractFloat, δ::AbstractFloat, ϵ::AbstractFloat)
```

Comment:
consider renaming `DualAveraging` into `NesterovDualAveraging`, and add a reference to 

- Hoffman, M. D., & Gelman, A. (2014). The No-U-Turn Sampler: adaptively setting path lengths in Hamiltonian Monte Carlo. Journal of Machine Learning Research, 15(1), 1593-1623.
- Nesterov, Y. (2009). Primal-dual subgradient methods for convex problems. Mathematical programming, 120(1), 221-259.

Just to make the terminology `dual averaging` more precise.